quickwit-config/src/server_config.rs

```diff
@@ -175,19 +147,35 @@ impl SearcherConfig {
     pub fn grpc_socket_addr(&self) -> anyhow::Result<SocketAddr> {
         socket_addr_from_str(format!(
             "{}:{}",
-            self.grpc_listen_address, self.grpc_listen_port
+            self.rest_listen_address,
+            self.rest_listen_port + 1
         ))
     }
 
     pub fn discovery_socket_addr(&self) -> anyhow::Result<SocketAddr> {
         socket_addr_from_str(format!(
             "{}:{}",
-            self.discovery_listen_address, self.discovery_listen_port
+            self.rest_listen_address,
+            self.rest_listen_port + 2
         ))
     }
 
     pub fn peer_socket_addrs(&self) -> anyhow::Result<Vec<SocketAddr>> {
-        self.peer_seeds.iter().map(socket_addr_from_str).collect()
+        let peer_socket_addrs: Vec<SocketAddr> = self
+            .peer_seeds
+            .iter()
+            .flat_map(|peer_seed| {
```

Comment:
De we want to log each address that does not resolve?

Reply:
yes probably.

Reply:
So here we force the peer to use the same listen port as "self".
It will be painful when we want to test stuff.

I suggest we all putting a port in here. 
That port would be interpreted as the baseport.

It feels lame since, the first way this address will be used is the UDP swim stuff.
We could do 
base port = REST port = UDP Port. 

(Technically we could do GRPC  on the base port too and route based on the application type.)